test/apps/iodemo/io_demo.cc

```diff
@@ -153,31 +153,68 @@ class IoDemoRandom {
     template <typename T>
     static inline T rand(T min = std::numeric_limits<T>::min(),
                          T max = std::numeric_limits<T>::max() - 1) {
-        _seed = (_seed * _A + _C) & _M;
+        return rand(_seed, min, max);
+    }
+
+    template <typename T>
+    static inline T rand(unsigned &seed, T min = std::numeric_limits<T>::min(),
+                         T max = std::numeric_limits<T>::max() - 1) {
+        seed = (seed * _A + _C) & _M;
         /* To resolve that LCG returns alternating even/odd values */
         if (max - min == 1) {
-            return (_seed & 0x100) ? max : min;
+            return (seed & 0x100) ? max : min;
         } else {
-            return T(_seed) % (max - min + 1) + min;
+            return T(seed) % (max - min + 1) + min;
         }
     }
 
-    static inline void fill(void *buffer, size_t size) {
-        size_t tail_count = size & sizeof(uint64_t);
+    static inline void fill(unsigned &seed, void *buffer, size_t size) {
+        size_t tail_count = size & (sizeof(uint64_t) - 1);
```

Comment:
body_count = size / sizeof(uint64_t)

Reply:
than `tail_count = size -  body_count`?

Reply:
could be
or `tail_count = size % sizeof(uint64_t)` and trust gcc

Reply:
using `body_count` requires extra multiplication, IMHO better use `&` vs `%` in this case, i'm not sure if compiler take into account pow of 2 for `%`